### PR TITLE
How to deploy on production

### PR DESCRIPTION
**Pull request by marmoure**
Adds instructions on how to deploy to [petal](https://petal.evolvedbinary.com/), and also adds the instruction on how to deploy the backend API.

### CODE REVIEW

File: packages/prosemirror-lwdita-backend/README.md [hunk 1]
```diff
@@ -1,86 +1,246 @@
...
+$ sudo ufw allow "Nginx Full"
+$ sudo ufw allow "OpenSSH"
+$ sudo ufw enable
+```
 
-# Usage
-Build the code
+### 1. Build the backend
 ```shell
 $ yarn run build
```
Comment (marmoure) [inline packages/prosemirror-lwdita-backend/README.md:57]: @adamretter the build script does not build the backend, I suggest we create a new build script to instruct the user to cd in to the backend.

File: packages/prosemirror-lwdita-backend/README.md [hunk 2]
```diff
@@ -1,86 +1,246 @@
...
+### 6. Configure nginx for the Backend
+The Backend is a Node.js application, as it is not recommened to directly expose Node.js to the Web, we will place the nginx Web Server in front of it.
+
+We will create a website which initially is service by nginx using just HTTP to revesere-proxy requests to our Node.js application. Later we will use certbot to generate a TLS/SSL certificate from Let's Encrypt, and reconfigure nginx to serve it using HTTPS.
+
+Create a new file in `/etc/nginx/sites-available/` named `petal.evolvedbinary.com.conf` with the following content:
+```
+server {
+
+    listen 80;
```
Comment (marmoure) [inline packages/prosemirror-lwdita-backend/README.md:111]: @adamretter This will mean that the backend will not be using `https`.

**Review by marmoure — COMMENTED** [review]
The backend build instructions should be revised.

File: packages/prosemirror-lwdita-demo/README.md [hunk 3]
```diff
@@ -14,3 +14,100 @@ yarn start:demo
 
 This will start a demo on `http://localhost:1234`.
 If this port is already in use, `parcel` will assign a random port that you can see in the terminal logs.
+
+## Installing
+Here are some simple instructions if you wish to install and host a [Petal Demo website](https://petal.evolvedbinary.com).
+
+Our instructions are for the following suggested Environment:
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-demo/README.md:21]: "Environment" => "environment"

File: packages/prosemirror-lwdita-demo/README.md [hunk 4]
```diff
@@ -14,3 +14,100 @@ yarn start:demo
...
+### 5. Have nginx load your new config
+```shell
+$ sudo systemctl reload nginx
+```
+
+### 6. Check the HTTP website is available
+Visit [http://petal.evolvedbinary.com](https://petal.evolvedbinary.com) from a different computer and make sure you see the Petal Demo website.
+
+### 7. Switch your Petal Demo website to HTTPS
+ We will use certbot to generate a TLS/SSL certificate from Let's Encrypt, and reconfigure nginx to serve it using HTTPS.
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-demo/README.md:88]: Remove space before "We ..."

File: packages/prosemirror-lwdita-backend/README.md [hunk 5]
```diff
@@ -28,24 +19,187 @@ It sets following values:
...
+```
 
+If you also want to enable a simple firewall on Ubuntu:
+```shell
+$ sudo ufw allow "Nginx Full"
+$ sudo ufw allow "OpenSSH"
+$ sudo ufw enable
+```
+
+### 1. Build the backend
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-backend/README.md:67]: Capitalize heading.

File: packages/prosemirror-lwdita-backend/README.md [hunk 6]
```diff
@@ -28,24 +19,187 @@ It sets following values:
...
 ```shell
 $ yarn run build
 ```
 
-Start the server:
+### 2. Deploy the Backend distribution files to the Server
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-backend/README.md:72]: Capitalize heading.

File: packages/prosemirror-lwdita-backend/README.md [hunk 7]
```diff
@@ -28,24 +19,187 @@ It sets following values:
...
+```
+
+Create a new file named `petal-api.conf` in `/etc/nginx/conf.d/` with the following content:
+```
+upstream petal-api {
+  server localhost:3000;
+}
+```
 
+### 7. Enable the nginx for your Petal Demo website
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-backend/README.md:156]: Capitalize heading.

File: packages/prosemirror-lwdita-backend/README.md [hunk 8]
```diff
@@ -28,24 +19,187 @@ It sets following values:
...
 ```shell
-$ yarn run start
+$ sudo ln -s /etc/nginx/sites-available/petal.evolvedbinary.com.conf /etc/nginx/sites-enabled/
 ```
 
+### 8. Have nginx load your new config
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-backend/README.md:161]: Capitalize heading.

File: packages/prosemirror-lwdita-backend/README.md [hunk 9]
```diff
@@ -28,24 +19,187 @@ It sets following values:
...
+```shell
+$ sudo systemctl reload nginx
+```
+
+### 9. Check the HTTP API is available
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-backend/README.md:166]: Capitalize heading.

File: packages/prosemirror-lwdita-backend/README.md [hunk 10]
```diff
@@ -28,24 +19,187 @@ It sets following values:
...
+Visit [http://petal.evolvedbinary.com/api/github](https://petal.evolvedbinary.com/api/github) from a different computer and make sure you see the response `Github API`.
+
+### 10. Switch your Petal Demo website to HTTPS
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-backend/README.md:169]: Capitalize heading.

File: packages/prosemirror-lwdita-backend/README.md [hunk 11]
```diff
@@ -28,24 +19,187 @@ It sets following values:
...
+- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+1: No redirect - Make no further changes to the webserver configuration.
+2: Redirect - Make all requests redirect to secure HTTPS access. Choose this for
+new sites, or if you're confident your site works on HTTPS. You can undo this
+change by editing your web server's configuration.
+- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+Select the appropriate number [1-2] then [enter] (press 'c' to cancel): 2
+ ```
+
+### 11. Check the HTTPS API is available
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-backend/README.md:193]: Capitalize heading.

File: packages/prosemirror-lwdita-demo/README.md [hunk 12]
```diff
@@ -14,3 +14,100 @@ yarn start:demo
...
+```
+
+If you also want to enable a simple firewall on Ubuntu:
+```shell
+$ sudo ufw allow "Nginx Full"
+$ sudo ufw allow "OpenSSH"
+$ sudo ufw enable
+```
+
+### 1. Build the demo
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-demo/README.md:42]: Capitalize heading.

File: packages/prosemirror-lwdita-demo/README.md [hunk 13]
```diff
@@ -14,3 +14,100 @@ yarn start:demo
...
+```shell
+$ yarn build:demo
+```
+
+### 2. Deploy the Petal Demo distribution files to the Web Server
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-demo/README.md:47]: Capitalize heading.

File: packages/prosemirror-lwdita-demo/README.md [hunk 14]
```diff
@@ -14,3 +14,100 @@ yarn start:demo
...
+Copy the all of the files from the `dist` folder of the Petal Demo to the folder on your nginx server for your website, we are using `/www-data/petal.evolvedbinary.com`. You will need to make sure that folder exists on the server before running this command.
+
+```shell
+$ scp dist/* ubuntu@petal.evolvedbinary.com:/www-data/petal.evolvedbinary.com/
+```
+
+### 3. Configure nginx for your Petal Demo website
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-demo/README.md:54]: Capitalize heading.

File: packages/prosemirror-lwdita-demo/README.md [hunk 15]
```diff
@@ -14,3 +14,100 @@ yarn start:demo
...
+
+    charset utf-8;
+    access_log /var/log/nginx/petal.evolvedbinary.com_ssl_access.log;
+
+    root /www-data/petal.evolvedbinary.com;
+    index index.html;
+}
+```
+
+### 4. Enable the nginx for your Petal Demo website
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-demo/README.md:74]: Capitalize heading.

File: packages/prosemirror-lwdita-demo/README.md [hunk 16]
```diff
@@ -14,3 +14,100 @@ yarn start:demo
...
+```shell
+$ sudo ln -s /etc/nginx/sites-available/petal.evolvedbinary.com.conf /etc/nginx/sites-enabled/
+```
+
+### 5. Have nginx load your new config
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-demo/README.md:79]: Capitalize heading.

File: packages/prosemirror-lwdita-demo/README.md [hunk 17]
```diff
@@ -14,3 +14,100 @@ yarn start:demo
...
+```shell
+$ sudo systemctl reload nginx
+```
+
+### 6. Check the HTTP website is available
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-demo/README.md:84]: Capitalize heading.

File: packages/prosemirror-lwdita-demo/README.md [hunk 18]
```diff
@@ -14,3 +14,100 @@ yarn start:demo
...
+Visit [http://petal.evolvedbinary.com](https://petal.evolvedbinary.com) from a different computer and make sure you see the Petal Demo website.
+
+### 7. Switch your Petal Demo website to HTTPS
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-demo/README.md:87]: Capitalize heading.

File: packages/prosemirror-lwdita-demo/README.md [hunk 19]
```diff
@@ -14,3 +14,100 @@ yarn start:demo
...
+- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+1: No redirect - Make no further changes to the webserver configuration.
+2: Redirect - Make all requests redirect to secure HTTPS access. Choose this for
+new sites, or if you're confident your site works on HTTPS. You can undo this
+change by editing your web server's configuration.
+- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+Select the appropriate number [1-2] then [enter] (press 'c' to cancel): 2
+ ```
+
+### 8. Check the HTTPS website is available
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-demo/README.md:111]: Capitalize heading.

**Review by plutonik-a — COMMENTED** [review]
@marmoure @adamretter I have found some typos in https://github.com/evolvedbinary/prosemirror-lwdita/pull/439, which already has been merged.

File: packages/prosemirror-lwdita-backend/README.md [hunk 20]
```diff
@@ -1,86 +1,246 @@
-# prosemirror-lwdita-backend
-API Module for ProseMirror-lwDITA
+# LwDITA ProseMirror Backend
+This module provided an API for working with ProseMirror LwDITA.
```
[the remark this answers is not in the document]
Comment (adamretter) [reply packages/prosemirror-lwdita-backend/README.md:2]: Thanks @plutonik-a I have fixed this in the `main` branch.

File: packages/prosemirror-lwdita-backend/README.md [hunk 21]
```diff
@@ -1,86 +1,246 @@
...
+  ```shell
+  $ yarn run start
+  ```
+
+## Installing
+Here are some simple instructions if you wish to install and host an [LwDITA ProseMirror Backend](https://petal.evolvedbinary.com/api).
+
+If you wish, the backend can be deployed on the same server that runs the [ProseMirror LwDITA Demo](../prosemirror-lwdita-demo/README.md).
+
+Our instructions are for the following suggested Environment:
```
[the remark this answers is not in the document]
Comment (adamretter) [reply packages/prosemirror-lwdita-backend/README.md:42]: Thanks @plutonik-a I have fixed this in the `main` branch.

**Pull request merged**